CONTRIBUTING.md: Add some style guidelines

akgrant43 opened the pull request:
[ci skip]

akgrant committed:
CONTRIBUTING.md: Add some style guidelines

[ci skip]

fniephaus commented on CONTRIBUTING.md:
````diff
@@ -212,3 +212,30 @@ If you were working on a feature branch, do this:
+C function declarations should have the type on one line and the 
+function name and parameters on the following line, e.g.:
+
+```
````
Adding a *c* to the end of this line should enable C-syntax highlighting.

fniephaus commented on CONTRIBUTING.md:
```diff
@@ -212,3 +212,30 @@ If you were working on a feature branch, do this:
+
+## Tabs are 4 spaces
+
+The heading says it all.
```
I'm afraid the title doesn't answer the question *when to use* spaces/tabs? Are tabs banned?

nicolas-cellier-aka-nice replied:
A side note: the legacy platforms/win32 is using these settings:

8-space tabs stops
2-space indent

I don't want to create massive change of ownership for fixing that...
https://stackoverflow.com/questions/4112410/git-change-styling-whitespace-without-changing-ownership-blame

akgrant43 commented:
Hi Fabio & Nicolas,

Thanks for your feedback and my apologies for not responding, I've been caught up with other things the last couple of weeks.  I haven't forgotten and definitely not ignoring this.

Cheers,
Alistair

fniephaus commented:
No worries at all, there's no rush. Have a great weekend...

Fabio

akgrant committed:
CONTRIBUTING.md: formatting guidelines

- Enable C syntax highlightin in the example
- Update tabs recommendation (keep existing styles, move to tab stop of
  4 spaces)

Thanks to @fniephaus and @nicolas-cellier-aka-nice for their feedback.

akgrant43 commented:
There haven't been any more comments for a couple of weeks.  I'm not particularly happy with the section on tabs vs spaces, however it's always easier to criticise someone else's work than start with a blank slate, so take this as inspiration! :-)

[ci skip]
